docker/scripts/cuda/runtime/install-vllm.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -Eeu
+
+# installs vllm and dependencies in runtime stage
+#
+# Required environment variables:
+# - VLLM_REPO: vLLM git repository URL
+# - VLLM_COMMIT_SHA: vLLM commit SHA to checkout
+# - VLLM_PREBUILT: whether to use prebuilt wheel (1/0)
+# - VLLM_USE_PRECOMPILED: whether to use precompiled binaries (1/0)
+
+# shellcheck source=/dev/null
+source /opt/vllm/bin/activate
+
+# build list of packages to install
+INSTALL_PACKAGES=(
+  nixl
+  cuda-python
+  'huggingface_hub[hf_xet]'
+  /tmp/wheels/*.whl
+)
+
+# clone vllm repository
+git clone "${VLLM_REPO}" /opt/vllm-source
+git -C /opt/vllm-source config --system --add safe.directory /opt/vllm-source
+git -C /opt/vllm-source fetch --depth=1 origin "${VLLM_COMMIT_SHA}" || true
+git -C /opt/vllm-source checkout -q "${VLLM_COMMIT_SHA}"
+
+# detect if prebuilt wheel exists
+WHEEL_URL=$(pip install \
+  --no-cache-dir \
+  --no-index \
+  --no-deps \
+  --find-links "https://wheels.vllm.ai/${VLLM_COMMIT_SHA}/vllm/" \
+  --only-binary=:all: \
+  --pre vllm \
+  --dry-run \
+  --disable-pip-version-check \
+  -qqq \
+  --report - \
+  2>/dev/null | jq -r '.install[0].download_info.url')
+
+if [ "${VLLM_PREBUILT}" = "1" ]; then
+  if [ -z "${WHEEL_URL}" ]; then
+    echo "VLLM_PREBUILT set but no platform compatible wheel exists for: https://wheels.vllm.ai/${VLLM_COMMIT_SHA}/vllm/"
+    exit 1
+  fi
+  INSTALL_PACKAGES+=("${WHEEL_URL}")
+  rm /opt/warn-vllm-precompiled.sh
+else
+  if [ "${VLLM_USE_PRECOMPILED}" = "1" ] && [ -n "${WHEEL_URL}" ]; then
+    echo "Using precompiled binaries and shared libraries for commit: ${VLLM_COMMIT_SHA}."
+    export VLLM_USE_PRECOMPILED=1
+    export VLLM_PRECOMPILED_WHEEL_LOCATION="${WHEEL_URL}"
+    INSTALL_PACKAGES+=(-e /opt/vllm-source)
```

Comment:
Wont this be treated as a package rather than an argument because this will add it to the array? Why don't we just add it to the "${WHEEL_URL}" before installing and move adding that to `INSTALL_PACKAGES` later?